src/stratis_cli/_actions/_report_pool.py

```diff
@@ -0,0 +1,160 @@
+# limitations under the License.
+"""
+Pool report.
+"""
+
+# isort: STDLIB
+import subprocess
+from abc import ABC, abstractmethod
+
+from .._errors import StratisCliResourceNotFoundError
+from .._stratisd_constants import PoolIdType
+from ._connection import get_object
+from ._constants import TOP_OBJECT
+
+
+def _fetch_stopped_pools_property(proxy):
+    """
+    Fetch the StoppedPools property from stratisd.
+    :param proxy: proxy to the top object in stratisd
+    :return: a representation of stopped devices
+    :rtype: dict
+    :raises StratisCliEngineError:
+    """
+
+    # pylint: disable=import-outside-toplevel
+    from ._data import Manager
+
+    return Manager.Properties.StoppedPools.Get(proxy)
+
+
+def report_pool(uuid_formatter, selection, *, stopped=False):
+    """
+    List the specified information about pools.
+    """
+    klass = (
+        Stopped(uuid_formatter, selection)
+        if stopped
+        else Default(uuid_formatter, selection)
+    )
+    klass.report_pool()
+
+
+class Report(ABC):  # pylint: disable=too-few-public-methods
+    """
+    Handle reporting on a pool or pools.
+    """
+
+    def __init__(self, uuid_formatter, selection):
+        """
+        Initialize a List object.
+        :param uuid_formatter: function to format a UUID str or UUID
+        :param uuid_formatter: str or UUID -> str
+        :param selection: how to select the pool to display
+        :type selection: pair of str * object
+        """
+        self.uuid_formatter = uuid_formatter
+        self.selection = selection
+
+    @abstractmethod
+    def report_pool(self):
+        """
+        List the pools.
+        """
+
+
+class Default(Report):  # pylint: disable=too-few-public-methods
+    """
+    Report on pools that stratis list by default.
+    """
+
+    def report_pool(self):
+        """
+        Do pool reports.
+        """
+        # pylint: disable=import-outside-toplevel
+        from ._data import MODev, ObjectManager, devs, pools
+
+        proxy = get_object(TOP_OBJECT)
+
+        managed_objects = ObjectManager.Methods.GetManagedObjects(proxy, {})
+
+        (selection_type, selection_value) = self.selection
+        (selection_key, selection_value) = (
+            ("Uuid", selection_value.hex)
+            if selection_type == PoolIdType.UUID
+            else ("Name", selection_value)
+        )
+
+        (pool_object_path, _) = next(
+            pools(props={selection_key: selection_value})
+            .require_unique_match(True)
+            .search(managed_objects)
+        )
+
+        modevs = (
+            MODev(info)
+            for objpath, info in devs(props={"Pool": pool_object_path}).search(
+                managed_objects
+            )
+        )
+
+        # replace the lines below with the correct call to lsblk
+        blkdevs = [dev.Devnode() for dev in modevs]
+        subprocess.run(["lsblk", "-i"] + blkdevs, check=True)
```

Comment:
For this line bandit reports two errors: B603 and B607. We should either explicitly allow these using the ```nosec``` annotation _and_ a comment explaining why _or_ correct the code so that bandit doesn't report any error.

Reply:
Resolved; the B603 error is ignored due to the inputs coming from D-Bus.